src/priorityqueue.jl

```diff
@@ -306,10 +274,16 @@ function dequeue_pair!(pq::PriorityQueue)
     return x
 end
 
-function dequeue_pair!(pq::PriorityQueue, key)
+if isdefined(Base, :popat!)
+    Base.popat!(pq::PriorityQueue, key) = _popat!(pq, key)
+else
+    popat!(pq::PriorityQueue, key) = _popat!(pq, key)
+end
+
+function _popat!(pq::PriorityQueue, key)
```

Comment:
What do you think of varying the normal rule against using import?
```suggestion
if isdefined(Base, :popat!)  # We will overload if it is defined, else we define on our own
    import popat!
end

function popat!(pq::PriorityQueue, key)
```
I think it might be the lesser of two evils here?

Reply:
I like what you proposed in https://github.com/JuliaCollections/DataStructures.jl/pull/771#discussion_r751502217, but I don't have a strong preference here. Which two evils do you mean here?

Reply:
Evil 1: is `import Foo: bar` because it allowed unqualified overloads
Evil 2: is this code is kinda complex for what it it does.

I think yes let's conditionally import it